docs/docs/pages/style-spec/sources.md

```diff
@@ -0,0 +1,319 @@
+"geojson-lines": {
+    "type": "geojson",
+    "data": "./lines.geojson"
+}
+```
+<!--
+START GENERATED CONTENT:
+Content in this section is generated directly using the MapLibre Style
+Specification. To update any content displayed in this section, make edits to:
+https://github.com/maplibre/maplibre-gl-style-spec/blob/main/src/reference/v8.json.
+-->
+{{<Items headingLevel='3' entry={ref.source_geojson} section="geojson" />}}
+<!-- END GENERATED CONTENT -->
+
+{{
+<SDKSupportTable
+    supportItems={{
+        'basic functionality': {
+            js: '0.10.0',
+            android: '2.0.1',
+            ios: '2.0.0',
+            macos: '0.1.0'
+        },
+        clustering: {
+            js: '0.14.0',
+            android: '4.2.0',
+            ios: '3.4.0',
+            macos: '0.3.0'
+        },
+        'line distance metrics': {
+            js: '0.45.0',
+            android: '6.5.0',
+            ios: '4.4.0',
+            macos: '0.11.0'
+        }
+    }}
+/>
+}}
+
+## image
+
+An image source. The `"url"` value contains the image location.
+
+The `"coordinates"` array contains `[longitude, latitude]` pairs for the image corners listed in clockwise order: top left, top right, bottom right, bottom left.
+
+```json
+"image": {
+    "type": "image",
+    "url": "https://maplibre.org/maplibre-gl-js-docs/assets/radar.gif",
+    "coordinates": [
+        [-80.425, 46.437],
+        [-71.516, 46.437],
+        [-71.516, 37.936],
+        [-80.425, 37.936]
+    ]
+}
+```
+<!--
+START GENERATED CONTENT:
+Content in this section is generated directly using the MapLibre Style
+Specification. To update any content displayed in this section, make edits to:
+https://github.com/maplibre/maplibre-gl-style-spec/blob/main/src/reference/v8.json.
+-->
+{{<Items headingLevel='3' entry={ref.source_image} section="image" />}}
+<!-- END GENERATED CONTENT -->
+
+{{
+<SDKSupportTable
+    supportItems={{
+        'basic functionality': {
+            js: '0.10.0',
+            android: '5.2.0',
+            ios: '3.7.0',
+            macos: '0.6.0'
+        }
+    }}
+/>
+}}
+
+
+
+## video
+
+A video source. The `"urls"` value is an array. For each URL in the array, a video element [source](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source) will be created. To support the video across browsers, supply URLs in multiple formats.
+
+The `"coordinates"` array contains `[longitude, latitude]` pairs for the video corners listed in clockwise order: top left, top right, bottom right, bottom left.
+
+```json
+"video": {
+    "type": "video",
+    "urls": [
+        "https://static-assets.mapbox.com/mapbox-gl-js/drone.mp4",
+        "https://static-assets.mapbox.com/mapbox-gl-js/drone.webm"
+    ],
+    "coordinates": [
+        [-122.51596391201019, 37.56238816766053],
+        [-122.51467645168304, 37.56410183312965],
+        [-122.51309394836426, 37.563391708549425],
+        [-122.51423120498657, 37.56161849366671]
+    ]
+}
+```
+
+<!--
```

Comment:
Seems like there's a generated code that's coming from the v8.json file, can't we move all this docs into it and avoid this duplication of docs and effort?